### ISSUE TITLE
Upgrade hibernate version

### ISSUE THREAD

**Issue by antarus** [jhipster/generator-jhipster#8095]
Hello,

##### **Overview of the issue**

Would it be possible to upgrade hibernate to the minimum version 5.3.1.Final?

##### **Motivation for or Use Case**
When using java 10, this version is a prerequisite to compile.

##### **Reproduce the error**
create a project,

Change the version of java to java 10,
Disable the enforcer of the java version,
Add to the configuration of the maven-compiler -plugin
```
<fork>true</fork>
<compilerArgs>
  <arg>-J--add-modules</arg>
  <arg>-Jjava.xml.bind</arg>
</compilerArgs>
```

and add the dependencies
```
<dependency>
    <groupId>javax.xml.bind</groupId>
    <artifactId>jaxb-api</artifactId>
</dependency>
<dependency>
    <groupId>com.sun.xml.bind</groupId>
    <artifactId>jaxb-impl</artifactId>
  <version>2.2.11</version>
</dependency>
<dependency>
    <groupId>com.sun.xml.bind</groupId>
    <artifactId>jaxb-core</artifactId>
  <version>2.2.11</version>
</dependency>
```

##### **Related issues**
none

##### **Suggest a Fix**

upgrade hibernate to the minimum version 5.3.1.Final

##### **JHipster Version(s)**
5.1.0

##### **JHipster configuration**
```
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.mycompany.myapp",
      "nativeLanguage": "fr"
    },
    "jhipsterVersion": "5.1.0",
    "applicationType": "gateway",
    "baseName": "front",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "serverPort": "8080",
    "authenticationType": "oauth2",
    "cacheProvider": "hazelcast",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "postgresql",
    "prodDatabaseType": "postgresql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "clientFramework": "angularX",
    "useSass": true,
    "clientPackageManager": "yarn",
    "testFrameworks": [
      "gatling"
    ],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "fr",
    "languages": [
      "fr",
      "en"
    ]
  }
}
```

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**
none

thanks

**Comment by jdubois** [comment]
Hi @antarus !
We're using the version provided by Spring Boot, and as you can see at https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-dependencies/2.0.4.RELEASE/spring-boot-dependencies-2.0.4.RELEASE.pom it's currently fixed at `5.2.17.Final`.
As this is tested & validated by the Spring Boot team, I don't think we should change this ourselves. Of course, you can still upgrade it manually in your own `pom.xml` - if you do so, can you provide us some feedback and tell if it all works fine?

**Comment by jdubois** [comment]
Yes indeed -> let's just wait for the next release, I'm closing this as this will be automatic
